src/core/shutdown.c

```diff
@@ -203,20 +202,25 @@ int main(int argc, char *argv[]) {
         }
 
         (void) cg_get_root_path(&cgroup);
+        in_container = detect_container() > 0;
 
         use_watchdog = !!getenv("WATCHDOG_USEC");
 
-        /* lock us into memory */
+        /* Lock us into memory */
         mlockall(MCL_CURRENT|MCL_FUTURE);
 
+        /* Synchronize everything that is not written to disk yet at this point already. This is a good idea so that
+         * slow IO is processed here already and the final process killing spree is not impacted by processes
+         * desperately trying to sync IO to disk within their timeout. */
```

Comment:
Well, I was thinking you were *intentionally* doing the sync with the watchdog disabled, to avoid it spuriously causing failures should sync take too long.

It just took me a moment to scrutinize the code and see that the watchdog gets enabled transparently by `watchdog_ping()` if it hasn't been enabled yet.  My comment wasn't meant to be contrary to the change, more a statement regarding the subtlety of how the watchdog gets enabled and the advantage of the sync occurring before the first ping.